src/components/VirtualList/VirtualItem.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { TResource, Resource, IstioTypes } from './Config';
+import { PromisesRegistry } from '../../utils/CancelablePromises';
+import * as Renderer from './Renderer';
+
+type VirtualItemProps = {
+  item: TResource;
+  style: any;
+  className: string;
+  index: number;
+  config: Resource;
+};
+type VirtualItemState = {
+  health?: any;
+};
+
+export default class VirtualItem extends React.Component<VirtualItemProps, VirtualItemState> {
+  private promises = new PromisesRegistry();
+
+  constructor(props: VirtualItemProps) {
+    super(props);
+    this.state = { health: undefined };
+  }
+
+  componentDidMount() {
+    if (this.props.item['healthPromise']) {
+      this.onItemChanged(this.props.item);
+    }
+  }
+
+  componentDidUpdate(prevProps: VirtualItemProps) {
+    if (this.props.item['healthPromise'] && this.props.item['healthPromise'] !== prevProps.item['healthPromise']) {
+      this.onItemChanged(this.props.item);
+    }
+  }
+
+  componentWillUnmount() {
+    this.promises.cancelAll();
+  }
+  onItemChanged(item: TResource) {
```

Comment:
Is this new code or did you move it from somewhere else? If so, can we use `async/await` here? Seems bad adding new code with old-style promises.

Reply:
This was moved from ItemDescription of workloads, application, istio and services, it's not a new code, I have not problem about move to the new styles, is there any development guide in readme or example about how you want to deal with promises?

Reply:
I think you can check the examples on `Login.ts` to see what I mean :)